perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -5021,16 +5088,18 @@ public void recalculateMemberGroupStatusRecursively(PerunSession sess, Member me
 		if (saveStatuses) {
 			groupsManagerImpl.setIndirectGroupStatus(sess, member, group, newStatus);
 
-			if (newStatus.equals(MemberGroupStatus.EXPIRED)) {
-				getPerunBl().getAuditer().log(sess, new MemberExpiredInGroup(member, group));
-			} else if (newStatus.equals(MemberGroupStatus.VALID)) {
-				getPerunBl().getAuditer().log(sess, new MemberValidatedInGroup(member, group));
+			if (!newStatus.equals(previousStatus.get(group.getId()).get(member.getId()))) {
```

Comment:
I think it might be worth checking if the previousStatues contain a map for the group. It might happen due to a race condition. If there are no previous statuses for the groupId, it should also audit a message.

Reply:
Good point, added, thanks.